server/controllers/profile.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const Profile = require('../models/Profile');
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile
+// @desc Create profile
+// @access Public
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    // Checking for empty input
+    for (let key in req.body) {
+        if (req.body[key].trim() === "") {
+          res.status(406);
+          throw new Error("Invalid input, please do not send empty input(s)")
+        }
+      }
+
+    const profile = new Profile(req.body);
+
+    try {
+        await profile.save(); 
+        res.status(201).json({ success: { profile }});
+    } catch(error) {
+        res.status(400).json({ error });
+    }
+});
+
+// @route PATCH /profile/:id
+// @desc update profile
+// @access private
+exports.updateProfile = asyncHandler(async (req, res, next) => {
+    const id = req.params.id;
+    const updates = req.body;
+    const options = { new: true };
+
+    try {
+        const updatedProfile = await Profile.findByIdAndUpdate(id, updates, options);
+        res.status(200).json({ success: { updatedProfile }});
+    } catch (error) {
+        res.status(400).json({ error });
+    }
```

Comment:
Similar concerns about `try...catch` block here. Also, nice touch on the `new` option 💯